pkg/hostagent/agent.go

```diff
@@ -497,22 +478,35 @@ func (agent *HostAgent) runTickers(stopCh <-chan struct{}) {
 	}
 }
 
-func (agent *HostAgent) checkSyncProcessorsCompletionStatus(kubeClient *kubernetes.Clientset) {
+func (agent *HostAgent) checkSyncProcessorsCompletionStatus(stopCh <-chan struct{}) {
 	ticker := time.NewTicker(time.Second)
 	defer ticker.Stop()
 
-	for range ticker.C {
-		agent.indexMutex.Lock()
-		count := len(agent.completedSyncTypes)
-		agent.indexMutex.Unlock()
-
-		if count == 5 {
-			err := agent.removeTaint(os.Getenv("KUBERNETES_NODE_NAME"), "aci-containers-host/unavailable", kubeClient)
+	kubeClient := agent.env.(*K8sEnvironment).kubeClient
+	for {
+		select {
+		case <-ticker.C:
+			node, err := kubeClient.CoreV1().Nodes().Get(context.TODO(), os.Getenv("KUBERNETES_NODE_NAME"), metav1.GetOptions{})
 			if err != nil {
-				agent.log.Errorf("Failed to remove taint: %v", err)
 				continue
 			}
-			agent.taintRemoved.Store(true)
+			for _, taint := range node.Spec.Taints {
+				if taint.Key == "aci-containers-host/unavailable" {
+					agent.indexMutex.Lock()
```

Comment:
I think check should based on key as well as effect, We might have same key with different tolerations effect,
Also can we avoid hardcoded key value here?

Reply:
Thank you for the review Ashok. Made the changes.